### ISSUE TITLE
Server support

### ISSUE THREAD

**Issue by snej** [acmacalister/jetfire#17]
JetFire only supports client-side connections. Server connections are required for apps that want to do P2P messaging (such as mine.)

The necessary bits are:

- [ ] Running a listener socket to accept incoming TCP connections
- [ ] Parsing the HTTP request on an incoming connection
- [ ] Modifying the way messages are sent (i.e. not masking the contents)

You may be able to steal code or at least ideas from [PocketSocket's implementation](https://github.com/zwopple/PocketSocket/blob/master/PocketSocket/PSWebSocketServer.m) :wink:

**Comment by acmacalister** [comment]
That would be a cool add, wasn't something I had originally planned on. Since this would probably take some time to test and run through the autobahn suite, PR is welcomed if you have something going.

**Comment by snej** [comment]
To be honest, I'm not likely to get around to this because I'm using PocketSocket instead of JetFire now :( I like JetFire's super small codebase, but server support is a must-have for my project and PocketSocket already has it.

**Comment by acmacalister** [comment]
Makes sense. Hard to argue when something already works. :smile: Maybe down the road we will be able to get this feature added.
